tracing-subscriber/src/registry/mod.rs

```diff
@@ -68,7 +68,7 @@ pub trait WithExtensions<'a> {
 }
 
 pub trait LookupSpan<'a> {
-    type Data: SpanData<'a>;
+    type Data: SpanData<'a> + WithExtensions<'a>;
```

Comment:
Personally, don't think this should be a bound enforced by `LookupSpan`, I'd prefer it if types that wanted to use extensions had bounds like
```rust
where
    S: for<'a> LookupSpan<'a>,
    S::Data: for<'a> WithExtensions<'a>,
```
but maybe requiring it here makes the API less complex. IDK how useful making it opt-in separately is.

But, if `WithExtensions` is going to be _required_ by the `LookupSpan` trait, it should probably just be methods on `SpanData` rather than a separate trait...